components/DisplayArticlesList.jsx

```diff
@@ -7,20 +7,23 @@ import {
   Dimensions,
   TouchableOpacity,
 } from "react-native";
+import { useSelector } from "react-redux"
 
 const DisplayArticlesList = ({ navigation, article }) => {
+  const { credentials } = useSelector(state => state)
+
   return (
     <TouchableOpacity
       onPress={() => {
-        navigation.navigate("SingleArticle", { article: article });
+        if (credentials) {
+          navigation.navigate("SingleArticle", { article: article })
+        } else {
+          alert("HISS! Please sign in to read an article!")
```

Comment:
try open this in the app simulation and see first how alert-message looks and to see if the message is not to long

Reply:
Yea we tried it, and it looked good :)